docs/protocol-spec.md

```diff
@@ -2126,7 +2126,7 @@ the code-metadata fields near it), and provide those to containers for deploymen
 ### 4.5 Online wallet service
 
 
-This section describes the security design of a wallet service, which in this case is a node where end-users can register, move their key material to, and perform transactions through.
+This section describes the security design of a wallet service, which in this case is a node with which end-users can register, store their key material, and through which they can perform transactions.
```

Comment:
their key **material and** through which they can perform transactions.

no `,`